src/h5web/visualizations/shared/utils.ts

```diff
@@ -93,10 +85,16 @@ export function extendDomain(
   return [min - extension, max + extension];
 }
 
-export function isIndexAxisConfig(
-  config: AxisConfig
-): config is IndexAxisConfig {
-  return 'indexDomain' in config;
+export function getValueToIndexScale(
+  values: number[]
+): ScaleThreshold<number, number> {
+  const indices = range(values.length).map((i) => i);
+
+  const valueToIndexScale = scaleThreshold();
+  valueToIndexScale.domain(values.slice(1)); // First threshold (going to 0 to 1) should be for the second value. Scaling the first value should return at 0.
+  valueToIndexScale.range(indices);
+
+  return valueToIndexScale;
```

Comment:
A [scaleThreshold](https://github.com/d3/d3-scale#threshold-scales) separates a `range` into arbitrary subsets that are given by the `domain`.

About the `slice`: imagine that we have 5 abscissas `[-5, -2.5, 0, 2.5, 5]` that should be mapped to `[0, 1, 2, 3, 4]`. Using naively these arrays will lead to
![image](https://user-images.githubusercontent.com/42204205/96238559-e9ff3c80-0f9e-11eb-9a00-f084b90f0903.png)

which is not what we want as `scale(x)` with `x` in `[-5, -2.5]` will return `1`. This is to be expected as the domain gives the thresholds: the first threshold being `-5`, the switch from `0` to `1` happens at this value`.

We want instead the switch from `0` to `1` to happen at the **second value** `-2.5`  hence the `slice`:
![image](https://user-images.githubusercontent.com/42204205/96239005-6c87fc00-0f9f-11eb-8685-b3353c43d93f.png)

We might be able to tune this to change the tooltip behaviour in the `LineVis` but we would need to make assumptions on the abscissas array (namely it is regularly spaced and increasing) while the `scaleThreshold` stay generic. This is a point to discuss as one could argue that abscissas **should always** be regularly spaced and increasing !